pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -808,21 +808,31 @@ func (m *ManagerImpl) filterByAffinity(podUID, contName, resource string, availa
 	var withoutTopology []string
 	for d := range available {
 		// Since the same device may be associated with multiple NUMA Nodes. We
-		// need to be careful not to add each device to multiple lists. The
-		// logic below ensures this by breaking after the first NUMA node that
-		// has the device is encountered.
+		// need to be careful not to add each device to multiple lists. The logic
+		// below ensures this by loop throug all NUMA Nodes and select a target
+		// list from the three devices list defined above and then insert the device
+		// into this list. For NUMA node that has the device, this loop will break
+		// only when it found that this node meets the affinity requirement and
+		// set fromAffinity as target list. Otherwise it will keep looping and:
+		//    i: set nodeWithoutTopology as target list when targetList is nil and
```

Comment:
It seems a comment of mine got lost. Let me retry.
IIUC, the problem is that we should de-prioritize devices who are affine to more than one numa node. When filtering devices, we should prefer devices who are compatible with the hint (obviously), but which have the stricter NUMA affinity set.
IOW, having hint pointing to NUMACELL2 between dev1 affine to NUMACELL2 (affinity=1 numa cell) and dev2 affine to NUMACELL1 and NUMACELL2 (affinity= 2 numa cells), prefer dev1.
Is this a good summarization?